src/lib/CPAuth/config.ts

```diff
@@ -0,0 +1,107 @@
+import { IOAuth2Config } from 'lib/OAuth2/OAuth2';
+import { AppRoutes } from 'shared/constants/routes';
+
+let authority = window.config.cpAudience;
+if (!/http(s)?:\/\//.test(authority)) {
+  authority = `https://${authority}`;
+}
+/**
+ * This is derived from the audience because it is verified by the
+ * OIDC plugin, and it must be the same as the production, non-proxied
+ * authentication provider.
+ */
+let issuer = authority;
+if (issuer.includes('localhost')) {
+  issuer = window.config.audience.replace('api', 'dex');
+}
+
+export const defaultConfig: IOAuth2Config = {
+  authority,
+  clientId: 'fFlz7lckhWA0kIaW3fLIl8chFSs2wvW6',
+  clientSecret:
+    'PoioOqWKUndxVnbcRzlv59EgvwPVJQIdIlved143Uko0SjGJ7OprnecZQbab3WhH',
```

Comment:
I think we really gotta figure out a way to do this without a client secret...

We kinda need dex to support PKCE: https://github.com/dexidp/dex/pull/1652

That's the flow we're using with auth0 I think

Reply:
Great find with that PR!
But can we move forward with this PR until we get that sorted?

Reply:
Btw, I don't think this is a huge problem right now, as users still need to have access to the control plane they try to login on.

Take for example a customer installation. Try authenticating with the kubectl plugin. Dex will allow the authentication request, since the client ID and Secret are correct, but you won't actually be able to log in, since you're not registered in the customer's authentication configuration (e.g. active directory).